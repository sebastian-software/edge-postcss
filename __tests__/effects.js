import { compile } from "./core"

// ====================================================
// ================== EFFECTS =========================
// ====================================================

test("Magic Animations", () =>
  compile(".animation { animation-name: magic; }")
)

test("Will Change Compat", () =>
  compile(".scaled { will-change: width; }")
)

test("Easings", () =>
  compile(".snake { transition: all 600ms ease-in-sine; }")
)

test("Pleeease Filters", () =>
  compile(".box { filter: drop-shadow(16px 16px 20px blue); }")
)

test("Transform Shortcut", () =>
  compile(".transform { scale: 2; translate: 10px 20px; }")
)

