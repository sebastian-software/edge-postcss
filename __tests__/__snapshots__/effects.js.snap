// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Easings 1`] = `
".snake{
transition:all 600ms cubic-bezier(.47,0,.745,.715)
}"
`;

exports[`Magic Animations 1`] = `
"@keyframes magic{
0%{
opacity:1;
transform-origin:100% 200%;
transform:scale(1,1) rotate(0deg)
}

to{
opacity:0;
transform-origin:200% 500%;
transform:scale(0,0) rotate(270deg)
}
}

.animation{
animation-name:magic
}"
`;

exports[`Pleeease Filters 1`] = `
".box{
filter:url('data:image/svg+xml;
charset=utf-8,<svg xmlns=\\"http://www.w3.org/2000/svg\\"><filter id=\\"filter\\"><feGaussianBlur in=\\"SourceAlpha\\" stdDeviation=\\"20\\" /><feOffset dx=\\"17\\" dy=\\"17\\" result=\\"offsetblur\\" /><feFlood flood-color=\\"rgba(0,0,255,1)\\" /><feComposite in2=\\"offsetblur\\" operator=\\"in\\" /><feMerge><feMergeNode /><feMergeNode in=\\"SourceGraphic\\" /></feMerge></filter></svg>#filter');
-webkit-filter:drop-shadow(16px 16px 20px #00f);
filter:drop-shadow(16px 16px 20px #00f)
}"
`;

exports[`Transform Shortcut 1`] = `
".transform{
transform:scale3d(2,1,1) translate3d(10px,20px,0)
}"
`;

exports[`Will Change Compat 1`] = `
".scaled{
-webkit-backface-visibility:hidden;
backface-visibility:hidden;
will-change:width
}"
`;
